pkg/doc/verifiable/presentation_test.go

```diff
@@ -0,0 +1,397 @@
+		raw := &rawPresentation{}
+		require.NoError(t, json.Unmarshal([]byte(validPresentation), &raw))
+		raw.Proof = nil
+		bytes, err := json.Marshal(raw)
+		require.NoError(t, err)
+		_, err = NewPresentation(bytes)
+		require.Error(t, err)
+		require.Contains(t, err.Error(), "embedded proof is missing")
+	})
+}
+
+func TestValidateVP_RefreshService(t *testing.T) {
+	t.Run("accepts verifiable presentation with empty refresh service", func(t *testing.T) {
+		raw := &rawPresentation{}
+		require.NoError(t, json.Unmarshal([]byte(validPresentation), &raw))
+		raw.RefreshService = nil
+		bytes, err := json.Marshal(raw)
+		require.NoError(t, err)
+		_, err = NewPresentation(bytes)
+		require.NoError(t, err)
+	})
+
+	t.Run("test verifiable presentation with undefined id of refresh service", func(t *testing.T) {
+		raw := &rawPresentation{}
+		require.NoError(t, json.Unmarshal([]byte(validPresentation), &raw))
+		raw.RefreshService = &RefreshService{Type: "ManualRefreshService2018"}
+		bytes, err := json.Marshal(raw)
+		require.NoError(t, err)
+		_, err = NewPresentation(bytes)
+		require.Error(t, err)
+		require.Contains(t, err.Error(), "refreshService: id is required")
+	})
+
+	t.Run("test verifiable presentation with undefined type of refresh service", func(t *testing.T) {
+		raw := &rawPresentation{}
+		require.NoError(t, json.Unmarshal([]byte(validPresentation), &raw))
+		raw.RefreshService = &RefreshService{ID: "https://example.edu/refresh/3732"}
+		bytes, err := json.Marshal(raw)
+		require.NoError(t, err)
+		_, err = NewPresentation(bytes)
+		require.Error(t, err)
+		require.Contains(t, err.Error(), "refreshService: type is required")
+	})
+
+	t.Run("test verifiable presentation with invalid URL of id of credential schema", func(t *testing.T) {
+		raw := &rawPresentation{}
+		require.NoError(t, json.Unmarshal([]byte(validPresentation), &raw))
+		raw.RefreshService = &RefreshService{ID: "invalid URL", Type: "ManualRefreshService2018"}
+		bytes, err := json.Marshal(raw)
+		require.NoError(t, err)
+		_, err = NewPresentation(bytes)
+
+		require.Error(t, err)
+		require.Contains(t, err.Error(), "refreshService.id: Does not match format 'uri'")
+	})
+}
+
+func TestPresentation_Credentials(t *testing.T) {
+	t.Run("decode verifiable credential from presentation", func(t *testing.T) {
+		vp, err := NewPresentation([]byte(validPresentation))
+		require.NoError(t, err)
+
+		credsData, err := vp.Credentials()
+		require.NoError(t, err)
+		require.Len(t, credsData, 1)
+
+		// Decode the first verifiable credential
+		vc, err := NewCredential(credsData[0])
+		require.NoError(t, err)
+
+		// check some VC properties to double check that conversion is OK
+		require.Equal(t, "http://example.edu/credentials/1872", vc.ID)
+		require.Equal(t, []string{"VerifiableCredential", "AlumniCredential"}, vc.Type)
+	})
+
+	t.Run("extracts verifiable credentials from list", func(t *testing.T) {
```

Comment:
How is this test different from one above?

Reply:
Not really, they could be merged together.